test/e2e/nodeportlocal_test.go

```diff
@@ -262,12 +292,14 @@ func NPLTestMultiplePods(t *testing.T) {
 	antreaPod, err := testData.getAntreaPodOnNode(node)
 	r.NoError(err, "Error when getting Antrea Agent Pod on Node '%s'", node)
 
+	targetProtocols := sets.NewString(protocolToString(defaultProtocolTCP))
 	for _, testPodName := range testPods {
 		targetPorts := sets.NewInt(defaultTargetPort)
 		nplAnnotations, testPodIP := getNPLAnnotations(t, testData, r, testPodName)
 
 		checkNPLRulesForPod(t, testData, r, nplAnnotations, antreaPod, testPodIP, true)
 		validatePortsInAnnotation(t, r, nplAnnotations, defaultStartPort, defaultEndPort, targetPorts)
+		validateProtocolsInAnnotation(t, r, nplAnnotations, targetProtocols)
```

Comment:
maybe there should be a `validateAnnotation` function to combine these 2? it feels unnatural to split port checking and protocol checking, and maybe even incorrect? the protocols may be different for each target port.